python/cudf/cudf/core/column/lists.py

```diff
@@ -453,10 +454,18 @@ def contains(self, search_key: ScalarLike) -> ParentType:
             raise
         return res
 
-    def index(self, search_key: ScalarLike) -> ParentType:
-        search_key = cudf.Scalar(search_key)
+    def index(self, search_key: Union[ScalarLike, ColumnLike]) -> ParentType:
```

Comment:
This method needs documentation and examples. I'm not sure if I fully understand when `-1` is returned and when `None` is returned, from looking at the tests.